modules/core/services/user_service.go

```diff
@@ -65,41 +65,39 @@ func (s *UserService) GetPaginatedWithTotal(ctx context.Context, params *user.Fi
 }
 
 func (s *UserService) Create(ctx context.Context, data user.User) error {
-	if err := composables.CanUser(ctx, permissions.UserCreate); err != nil {
-		return err
-	}
-	if err := s.validator.ValidateCreate(ctx, data); err != nil {
+	err := composables.CanUser(ctx, permissions.UserCreate)
+	if err != nil {
 		return err
 	}
-	logger := composables.UseLogger(ctx)
-	tx, err := composables.BeginTx(ctx)
+	err = s.validator.ValidateCreate(ctx, data)
 	if err != nil {
 		return err
 	}
-	defer func() {
-		if err := tx.Rollback(ctx); err != nil {
-			logger.WithError(err).Error("failed to rollback transaction")
+
+	var createdUser user.User
+	err = composables.InTx(ctx, func(txCtx context.Context) error {
+		data, err = data.SetPassword(data.Password())
+		if err != nil {
+			return err
 		}
-	}()
-	createdEvent, err := user.NewCreatedEvent(ctx, data)
-	if err != nil {
+		created, err := s.repo.Create(ctx, data)
+		if err != nil {
+			return err
+		}
+		createdUser = created
 		return err
-	}
-	data, err = data.SetPassword(data.Password())
+	})
 	if err != nil {
 		return err
 	}
-	created, err := s.repo.Create(ctx, data)
+	createdEvent, err := user.NewCreatedEvent(ctx, data)
 	if err != nil {
 		return err
 	}
-
-	if err := tx.Commit(ctx); err != nil {
-		return err
-	}
-	createdEvent.Result = created
+	createdEvent.Result = createdUser
 	s.publisher.Publish(createdEvent)
-	return nil
+
+	return err
```

Comment:
You're checking for `err != nil` on line 94. Returning `nil` is the correct behaviour